packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1476,8 +1528,117 @@ class RenderPhysicalModel extends _RenderCustomClip<RRect> {
     super.debugFillProperties(description);
     description.add(new DiagnosticsProperty<BoxShape>('shape', shape));
     description.add(new DiagnosticsProperty<BorderRadius>('borderRadius', borderRadius));
-    description.add(new DoubleProperty('elevation', elevation));
-    description.add(new DiagnosticsProperty<Color>('color', color));
+  }
+}
+
+/// Creates a physical shape layer that clips its child to a [ShapeBorder].
+///
+/// A physical shape layer casts a shadow based on its [elevation].
+///
+/// See also:
+///
+/// * [RenderPhysicalModel], which is optimized for rounded rectangles and
+///   circles.
+class RenderPhysicalShape extends _RenderPhysicalModelBase<Path> {
+  /// Creates an arbitrary shape clip.
+  ///
+  /// The [color], [shape], and [textDirection] are required.
+  ///
+  /// The [clipper], [elevation], [color] and [shadowColor] must
+  /// not be null.
+  RenderPhysicalShape({
+    RenderBox child,
+    @required CustomClipper<Path> clipper,
+    double elevation: 0.0,
+    @required Color color,
+    Color shadowColor: const Color(0xFF000000),
+    @required TextDirection textDirection,
+  }) : assert(clipper != null),
+       assert(elevation != null),
+       assert(color != null),
+       assert(shadowColor != null),
+       _textDirection = textDirection,
+       super(
+         child: child,
+         elevation: elevation,
+         color: color,
+         shadowColor: shadowColor,
+         clipper: clipper,
+       );
+
+  /// The text direction to use for getting the outer path for [shape].
+  ///
+  /// [ShapeBorder]s can depend on the text direction (e.g having a "dent"
+  /// towards the start of the shape).
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
```

Comment:
as far as i can tell you're not using this.

Reply:
Oops, leftovers from when this used ShapedBorder, removed.